pkg/network/usm/usm_http2_monitor_test.go

```diff
@@ -1103,6 +1103,36 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 				}: 2,
 			},
 		},
+		{
+			name: "validate CONTINUATION frame support",
+			// Testing the scenario in which part of the data frame header is sent using CONTINUATION frame.
+			// Currently, we do not support CONTINUATION frames, therefore, we expect to capture only the first
+			// part of the message.
+			messageBuilder: func() [][]byte {
+				preFixTestHeaders, sufFixTestHeaders := splitTestHeaders(2)
+				prFixHeadersFrame, err := usmhttp2.NewHeadersFrameMessage(usmhttp2.HeadersFrameOptions{
+					Headers: preFixTestHeaders,
+				})
+				require.NoError(t, err, "could not create headers frame")
+
+				sufFixHeadersFrame, err := usmhttp2.NewHeadersFrameMessage(usmhttp2.HeadersFrameOptions{
+					Headers: sufFixTestHeaders,
+				})
+				require.NoError(t, err, "could not create headers frame")
+
+				headersFrame := newFramer().writeRawHeaders(t, 1, false, prFixHeadersFrame).bytes()
+				continuationFrame := newFramer().writeRawContinuation(t, 1, true, sufFixHeadersFrame).bytes()
+				dataFrameSecond := newFramer().writeData(t, 1, true, emptyBody).bytes()
```

Comment:
you should be using the same framers, as the indexing between the headers and the continuation should match